git/git.go

```diff
@@ -43,6 +43,18 @@ type Ref struct {
 	Sha  string
 }
 
+// MatchesNameAndType returns whether or not the receiving *Ref "r" has the same
+// name and type as the "other" *Ref. Nil *Refs are invalid values and thus do
+// not possess equality.
+func (r *Ref) MatchesNameAndType(other *Ref) bool {
```

Comment:
Thanks for bringing this up. I can't think of any legal situations where this would/could happen off-hand, so I simplified the behavior in https://github.com/github/git-lfs/pull/1595/commits/cf8fac17aeb2c0c64b3c413d72af3f06019108a6.